File: employee360-frontend/src/types/feedback.ts
import type { FeedbackType } from './enums';

// ─── Feedback Response ──────────────────────────────────────────────
export interface Feedback {
    id: number;
    fromEmployeeName: string;
    fromEmployeeId: number;
    toEmployeeName: string;
    toEmployeeId: number;
    projectName: string | null;
    type: FeedbackType;
    content: string;
    rating: number;
    isAnonymous: boolean;
    createdAt: string; // ISO datetime
}

// ─── Request ────────────────────────────────────────────────────────
export interface FeedbackRequest {
    toEmployeeId: number;
    projectId?: number;
    type: FeedbackType;
    content: string;
    rating?: number;
    isAnonymous?: boolean;
}

File: employee360-backend/src/main/java/com/kpmg/employee360/entity/Feedback.java
package com.kpmg.employee360.entity;

import com.kpmg.employee360.enums.FeedbackType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "feedbacks", indexes = {
        @Index(name = "idx_fb_to_employee", columnList = "to_employee_id"),
        @Index(name = "idx_fb_from_employee", columnList = "from_employee_id")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Feedback {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "from_employee_id", nullable = false)
    private Employee fromEmployee;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "to_employee_id", nullable = false)
    private Employee toEmployee;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id")
    private Project project;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FeedbackType type;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    private Integer rating;

    @Column(nullable = false)
    @Builder.Default
    private Boolean isAnonymous = false;

    @CreationTimestamp
    private LocalDateTime createdAt;
}
